plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ScanQueryPageSource.java

```diff
@@ -309,9 +323,11 @@ private Decoder createDecoder(String path, Type type)
         }
         if (type instanceof RowType) {
             RowType rowType = (RowType) type;
+            Map<String, ElasticsearchColumnHandleProperty> children = handleProperty.getChildren();
 
             List<Decoder> decoders = rowType.getFields().stream()
-                    .map(field -> createDecoder(appendPath(path, field.getName().get()), field.getType()))
+                    .map(field -> createDecoder(appendPath(path, field.getName().get()), field.getType(),
+                            children.get(field.getName().get())))
```

Comment:
Can we add a test where nested fields have mixed casing? so that we know they're found during this lookup

Reply:
By mixed casing you're talking about the field name? 

`camelCase` for one field and `underscore_casing` for another?